src/libraries/System.Text.Json/gen/src/System.Text.Json.SourceGeneration.csproj

```diff
@@ -0,0 +1,47 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <CLSCompliant>false</CLSCompliant>
+    <Nullable>enable</Nullable>
+    <!-- Suppress warning XML comment has cref attribute that could not be resolved -->
+    <NoWarn>CS1574</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <DefineConstants>$(DefineConstants);BUILDING_SOURCE_GENERATOR</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisCSharpWorkspacesVersion)" PrivateAssets="all" />
+
+    <PackageDestination Include="analyzers\dotnet\cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\..\Common\JsonIgnoreCondition.cs" Link="Common\System\Text\Json\Serialization\JsonIgnoreCondition.cs" />
+    <Compile Include="..\..\Common\JsonNumberHandling.cs" Link="Common\System\Text\Json\Serialization\JsonNumberHandling.cs" />
+    <Compile Include="ClassType.cs" />
+    <Compile Include="CollectionType.cs" />
+    <Compile Include="IsExternalInit.cs" />
+    <Compile Include="JsonSourceGenerator.cs" />
+    <Compile Include="JsonSourceGeneratorHelper.cs" />
+    <Compile Include="JsonSerializableSyntaxReceiver.cs" />
+    <Compile Include="JsonSourceGeneratorHelper.Generate.cs" />
+    <Compile Include="ObjectConstructionStrategy.cs" />
+    <Compile Include="PropertyMetadata.cs" />
+    <Compile Include="Reflection\AssemblyWrapper.cs" />
+    <Compile Include="Reflection\TypeExtensions.cs" />
+    <Compile Include="Reflection\FieldInfoWrapper.cs" />
+    <Compile Include="Reflection\ConstructorInfoWrapper.cs" />
+    <Compile Include="Reflection\CustomAttributeDataWrapper.cs" />
+    <Compile Include="Reflection\MemberInfoWrapper.cs" />
+    <Compile Include="Reflection\MetadataLoadContextInternal.cs" />
+    <Compile Include="Reflection\MethodInfoWrapper.cs" />
+    <Compile Include="Reflection\ParameterInfoWrapper.cs" />
+    <Compile Include="Reflection\PropertyInfoWrapper.cs" />
+    <Compile Include="Reflection\ReflectionExtensions.cs" />
+    <Compile Include="Reflection\RoslynExtensions.cs" />
+    <Compile Include="Reflection\TypeWrapper.cs" />
```

Comment:
We're creating our own reflection-based implementation using ISymbol as the data source?  That seems much more generic than just for S.T.J... do we need our own internal copy?  Should this be a separate library this depends on?

Reply:
It's generic enough to be a separate library. It's just implemented internally in this project because we haven't found another consumer yet.

@davidfowl started the work in a uController prototype a while ago. David - is packaging this in a library a priority for you sometime soon?

Reply:
It should be separated but it was incomplete.

Reply:
I think we’d want to better understand how dependencies work in analyzers before creating new modules that are potentially used by multiple analyzers.  It could cause new issues.  Definitely something to consider.